main.cpp

```diff
@@ -373,17 +386,17 @@ struct Point
         return *this;
     }
 
-    Point& multiply(const FloatType& ft)
+    Point& multiply(const Numeric<float>& ft)
     {
        return multiply(static_cast<float>(ft));
     }
 
-    Point& multiply(const IntType& it)
+    Point& multiply(const Numeric<int>& it)
     {
        return multiply(static_cast<float>(it));
     }
 
-    Point& multiply(const DoubleType& dt)
+    Point& multiply(const Numeric<double>& dt)
     {
        return multiply(static_cast<float>(dt));
     }
```

Comment:
these functions can be eliminated entirely.
you can cast the argument being passed in to a `float`, which will call `multiply(float)`
- [ ] revise